### PR TITLE
Update to Nock 4K

### PR DESCRIPTION
**Pull request by cmarcelo**

### CODE REVIEW

**Comment by lukechampine** [comment]
Thanks for doing this!

I was surprised to discover recently that Nock 4K is somewhat controversial...or at least, ~fyr doesn't like it. So I'm wondering if maybe Nock 4K and Nock 5K can live alongside each other. Maybe in separate subfolders, or on different branches. What do you think?

**Comment by cmarcelo** [comment]
> I was surprised to discover recently that Nock 4K is somewhat controversial...or at least, ~fyr doesn't like it. So I'm wondering if maybe Nock 4K and Nock 5K can live alongside each other. Maybe in separate subfolders, or on different branches. What do you think?

_If_ we do want to keep both, there is significant reuse that I'm thinking about simply keeping them in the same package as two separate Nock functions: `Nock4()` and `Nock5()`. Perhaps make an alias for one of them as `Nock()`. What do you think?

**Comment by lukechampine** [comment]
That's true. Ok, let's make `Nock()` Nock 4K (as it currently is in this PR) and also add a `Nock5K()` function.

**Comment by cmarcelo** [comment]
There's Nock4 and Nock5, and Nock just calls Nock4. Also made other changes along the way.

@lukechampine could you take another look?

File: nock.go [hunk 1]
```diff
@@ -66,56 +66,175 @@ func fas(i int, n Noun) Noun {
...
+func tar5(sub, form Noun) Noun {
+	// Distribution rule.
+	// *[a [b c] d]        [*[a b c] *[a d]]
 	if form.Head().IsCell() {
-		return Cell(tar(sub, form.Head()), tar(sub, form.Tail()))
+		return Cell(tar5(sub, form.Head()), tar5(sub, form.Tail()))
 	}
 	inst, arg := form.Head(), form.Tail()
-	return map[int]func() Noun{
+	switch inst.Num() {
```
Comment (lukechampine) [inline nock.go:94]: I guess the map literal approach was a little unsafe... :P

**Comment by lukechampine** [comment]
ok, LGTM! I'll send your planet over shortly :)

**Pull request merged**